openedx-unsupported/frontend-app-payment
# refactor: Payment form for Stripe and CyberSource

[REV-3001](https://2u-internal.atlassian.net/browse/REV-3001).

Refactoring the Stripe component so that it becomes of sibling of the PaymentForm (used for CyberSource) removing one layer of the component hierarchy. Also took out the submit button to its own component so that it can be used for both CyberSource form and Stripe form.

Made a successful test payment with this refactor - 
<img width="1157" alt="Screen Shot 2022-10-05 at 1 50 06 PM" src="https://user-images.githubusercontent.com/13632680/194130014-0f780abb-f8c8-4a9c-a44a-e6ccfd7eec56.png">

**This PR:**
- Fixes the temp `true` on the `enableStripePaymentProcessor` flag value to use the actual waffle flag value from ecommerce Basket API.
- Moves the place order button to its own component for reusability in both CyberSource and Stripe payment forms.
- Moves the `StripeCardPayment` component to be its own component separate from `PaymentForm` (since it has its own form, it did not need to be in the PaymentForm component, removing one unnecessary child --> `PaymentForm` and `StripeCardPayment` will be siblings).
- Renames the `StripeCardPayment` component to `StripePaymentForm`

**How to test this locally:**
1. Checkout this branch
2. Make sure you run `make dev.down.frontend-app-payment` in your `/devstack` directory
3. On `frontend-app-payment` run `npm ci` + `npm start`
4. Enroll in a course and upgrade
5. Billing address form should load, as well as the Stripe card elements
6. Make a purchase and see the receipt page
